src/python/nxt_python_asgi.c

```diff
@@ -610,6 +613,17 @@ nxt_py_asgi_create_http_scope(nxt_unit_request_info_t *req)
         return NULL;
     }
 
+    prefix = app_target->prefix;
+    path_length = r->path_length;
+    p = nxt_unit_sptr_get(&r->path);
+    if (prefix.length > 0
+        && ((path_length > prefix.length && p[prefix.length] == '/')
+            || path_length == prefix.length)
```

Comment:
Maybe you could combine this test and the `nxt_memcmp()` one with the following:

```c
$ grepc nxt_str_start
./src/nxt_string.h:147:
#define nxt_str_start(s, p, _length)                                          \
    (((s)->length >= _length) && (nxt_memcmp((s)->start, p, _length) == 0))
```
So something like:
```c
nxt_str_start(p, prefix, prefix.start, prefix.length)
```
might help (but maybe not, and maybe it just complicates it more).  Please check.

Reply:
`nxt_str_start` won't work, because `p` is a `char *`, not an `nxt_str_t`.